admin-dev/themes/new-theme/js/app/pages/permission/permission.vue

```diff
@@ -0,0 +1,184 @@
+            :level-depth="1"
+            :max-level-depth="4"
+            :permission="permission"
+            :permission-id="permissionId"
+            :permission-key="permissionKey"
+            :profile-permissions.sync="profileDataPermissions"
+            :employee-permissions="employeePermissions"
+            :parent="permission.children !== undefined"
+            :types="Object.keys(types)"
+            @sendRequest="sendRequest"
+          >
+          </row>
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+  import Bulk from './components/bulk.vue';
+  import Row from './components/row.vue';
+
+  const $ = window.$;
+
+  export default {
+    name: 'permission',
+    components: {
+      Bulk,
+      Row,
+    },
+    props: {
+      title: {
+        type: String,
+        required: true,
+      },
+      emptyData: {
+        type: String,
+        required: true,
+      },
+      profileId: {
+        type: Number,
+        required: true,
+      },
+      messages: {
+        type: Object,
+        required: true,
+      },
+      updateUrl: {
+        type: String,
+        required: true,
+      },
+      permissionKey: {
+        type: String,
+        required: true,
+      },
+      types: {
+        type: Object,
+        required: true,
+      },
+      permissions: {
+        type: Object,
+        required: true,
+      },
+      profilePermissions: {
+        type: Object,
+        required: true,
+      },
+      employeePermissions: {
+        type: Object,
+        required: true,
+      },
+      canEdit: {
+        type: Boolean,
+        required: false,
+        default: false,
+      },
+    },
+    data() {
+      return {
+        profileDataPermissions: this.profilePermissions,
+      };
+    },
+    methods: {
+      /**
+       * Send ajax request to target url
+       */
+      sendRequest(data) {
+        data.profile_id = this.profileId;
+
+        $.ajax(
+          this.updateUrl,
+          {
+            method: 'POST',
+            data,
+          },
+        ).then((response) => {
+          if (response.success) {
+            window.showSuccessMessage(this.messages.success);
+            return;
+          }
+
+          window.showErrorMessage(this.messages.error);
+        }).catch(() => {
+          window.showErrorMessage(this.messages.error);
+        });
+      },
+      /**
+       * Update user permissions from bulk action
+       */
+      updateBulk(data) {
+        Object.keys(this.profileDataPermissions).forEach((key) => {
+          data.types.forEach((type) => {
+            this.profileDataPermissions[key][type] = data.status ? '1' : '0';
+          });
+        });
+
+        const params = {
+          expected_status: data.status,
+          permission: data.updateType,
+        };
+        params[this.permissionKey] = '-1';
+
+        this.sendRequest(params);
+      },
+    },
+  };
+</script>
+
+<style lang="scss">
```

Comment:
@PierreRambaud I'm a noob in vue.js but is it required 🤔 ? In-page CSS style ?

Reply:
It's a scss specific to this template.
With vuejs you're able to use three containers, `<template>`, `<script>` and `<style>`. for style you can `scoped` / `module` it and use a lang if you want to use `scss`, `less`, `sass`, etc.
It's really powerful if you want to manage your component style & actions in a single page. At least if you switch component, you don't need to change X files to remove js, css, & html.